filter/request.go

```diff
@@ -0,0 +1,62 @@
+package filter
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+//UpdateRequest will be called back when the request is received by the proxy.
+func (f *Filter) UpdateRequest(r *http.Request) {
+	requestLog := f.log.WithFields(logrus.Fields{"url": r.URL.String(), "action": "request", "source": r.RemoteAddr})
+	requestLog.Debug("Request")
+
+	u, _ := url.Parse(f.url)
+	r.URL.Host = u.Host
+	r.Host = u.Host
+	r.URL.Scheme = u.Scheme
+	data, err := httputil.DumpRequest(r, false)
+
+	if err != nil {
+		f.log.Error(fmt.Printf("Error"))
+	}
+
+	f.log.Debug(fmt.Sprintf("Request received\n %s", string(data)))
+
+	if r.Body != nil && len(f.request.Replace) > 0 {
+		s, err := f.readBody(r.Body, r.Header)
```

Comment:
May be this block shoud be move to a function replaceBody that take pointer to body, header, []Replacement and return the content length. Look to f.readBoy for an example?

That will allow to reuse-it for response too